pkg/daemon/ceph/client/command.go

```diff
@@ -67,62 +67,57 @@ func FinalizeCephCommandArgs(command string, args []string, configDir, clusterNa
 
 	// If the command should be run inside the toolbox pod, include the kubectl args to call the toolbox
 	if RunAllCephCommandsInToolbox {
-		toolArgs := []string{"-it", "exec", "rook-ceph-tools", "-n", clusterName, "--", command}
+		toolArgs := []string{"-it", "exec", "rook-ceph-tools", "-n", clusterInfo.Namespace, "--", command}
 		return Kubectl, append(toolArgs, args...)
 	}
 
-	// No need to append the args if it's the default ceph cluster
-	if clusterName == "ceph" && configDir == "/etc" {
-		return command, args
-	}
-
 	// Append the args to find the config and keyring
-	keyringFile := fmt.Sprintf("%s.keyring", username)
+	keyringFile := fmt.Sprintf("%s.keyring", clusterInfo.CephCred.Username)
 	configArgs := []string{
-		fmt.Sprintf("--cluster=%s", clusterName),
-		fmt.Sprintf("--conf=%s", CephConfFilePath(configDir, clusterName)),
-		fmt.Sprintf("--name=%s", username),
-		fmt.Sprintf("--keyring=%s", path.Join(configDir, clusterName, keyringFile)),
+		fmt.Sprintf("--cluster=%s", clusterInfo.Namespace),
+		fmt.Sprintf("--conf=%s", CephConfFilePath(configDir, clusterInfo.Namespace)),
+		fmt.Sprintf("--name=%s", clusterInfo.CephCred.Username),
+		fmt.Sprintf("--keyring=%s", path.Join(configDir, clusterInfo.Namespace, keyringFile)),
 	}
 	return command, append(args, configArgs...)
 }
 
 type CephToolCommand struct {
 	context     *clusterd.Context
+	clusterInfo *ClusterInfo
 	tool        string
-	clusterName string
 	args        []string
 	timeout     time.Duration
 	JsonOutput  bool
 	OutputFile  bool
 	authUser    string
 }
 
-func newCephToolCommand(tool string, context *clusterd.Context, clusterName string, args []string) *CephToolCommand {
+func newCephToolCommand(tool string, context *clusterd.Context, clusterInfo *ClusterInfo, args []string) *CephToolCommand {
 	return &CephToolCommand{
 		context:     context,
 		tool:        tool,
-		clusterName: clusterName,
+		clusterInfo: clusterInfo,
 		args:        args,
 		JsonOutput:  true,
 		OutputFile:  true,
 		authUser:    AdminUsername,
```

Comment:
Shouldn't this be the user from `clusterInfo`? It would be nice to have some unit test for this too.